clients/src/main/java/org/apache/kafka/common/utils/ChildFirstClassLoader.java

```
@@ -0,0 +1,131 @@
+import org.apache.kafka.common.KafkaException;
+
+
+/**
+ * A class loader that looks for classes and resources in a specified class path first, before delegating to its parent
+ * class loader.
+ */
+public class ChildFirstClassLoader extends URLClassLoader {
+    static {
+        ClassLoader.registerAsParallelCapable();
+    }
+
+    /**
+     * @param classPath Class path string
+     * @param parent    The parent classloader. If the required class / resource cannot be found in the given classPath,
+     *                  this classloader will be used to find the class / resource.
+     */
+    public ChildFirstClassLoader(String classPath, ClassLoader parent) {
+        super(classpath2URLs(classPath), parent);
+    }
+
+    static private URL[] classpath2URLs(String classPath) {
+        ArrayList<URL> urls = new ArrayList<>();
+        for (String path : classPath.split(File.pathSeparator)) {
+            if (path == null || path.trim().isEmpty())
+                continue;
+            File f = new File(path);
+
+            try {
+                if (path.endsWith("/*")) {
+                    File parent = new File(new File(f.getCanonicalPath()).getParent());
+                    if (parent.isDirectory()) {
+                        File[] files = parent.listFiles((dir, name) -> {
+                            String lower = name.toLowerCase(Locale.ROOT);
+                            return lower.endsWith(".jar") || lower.endsWith(".zip");
+                        });
+                        if (files != null) {
+                            for (File jarFile : files) {
+                                urls.add(jarFile.getCanonicalFile().toURI().toURL());
+                            }
+                        }
+                    }
+                } else if (f.exists()) {
+                    urls.add(f.getCanonicalFile().toURI().toURL());
+                }
+            } catch (IOException e) {
+                throw new KafkaException(e);
```

Throwing an exception instead of swallowing this silently.